test/jdk/javax/swing/JComboBox/EditableComboBoxPopupPos.java

```diff
@@ -0,0 +1,139 @@
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+
+public class EditableComboBoxPopupPos {
+    private static Robot robot;
+    private static JFrame frame;
+    private static JPanel panel;
+    private static JComboBox cb1, cb2;
+    private static String lafName;
+
+    private static final int BUTTON_OFFSET = 10;
+    private static final int POPUP_OFFSET = 5;
+
+    public static void main(String[] args) throws InterruptedException, InvocationTargetException {
+        try {
+            robot = new Robot();
+            robot.setAutoDelay(100);
+            robot.setAutoWaitForIdle(true);
+
+            for (UIManager.LookAndFeelInfo laf : UIManager.getInstalledLookAndFeels()) {
+                lafName = laf.getName().equals("CDE/Motif") ? "Motif" : laf.getName();
+                SwingUtilities.invokeAndWait(() -> {
+                    setLookAndFeel(laf);
+                    panel = new JPanel();
+                    String[] comboStrings = {"One", "Two", "Three"};
+                    cb1 = new JComboBox(comboStrings);
+                    cb1.setEditable(true);
+                    cb1.setBorder(BorderFactory.createTitledBorder(
+                            "Editable JComboBox"));
+
+                    cb2 = new JComboBox(comboStrings);
+                    cb2.setEditable(true);
+
+                    panel.add(cb1);
+                    panel.add(cb2);
+
+                    frame = new JFrame();
+                    frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+                    frame.add(panel);
+                    frame.pack();
+                    frame.setLocationRelativeTo(null);
+                    frame.setVisible(true);
+                });
+
+                // Change starting selection to check if the position of the
+                // first selection item is in the correct position on screen.
+                cb1.setSelectedIndex(1);
+                cb2.setSelectedIndex(1);
+
+                runTestOnComboBox(cb1);
+                runTestOnComboBox(cb2);
+
+                checkSelection(cb1, cb2);
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            SwingUtilities.invokeAndWait(() -> frame.dispose());
+        }
+    }
+
+    private static void setLookAndFeel(UIManager.LookAndFeelInfo laf) {
+        try {
+            UIManager.setLookAndFeel(laf.getClassName());
+        } catch (UnsupportedLookAndFeelException ignored){
+            System.out.println("Unsupported LookAndFeel: " + laf.getClassName());
+        } catch (ClassNotFoundException | InstantiationException |
+                IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void runTestOnComboBox(JComboBox cb) {
+        robot.mouseMove(cb.getLocationOnScreen().x + cb.getWidth()
+                - BUTTON_OFFSET, cb.getLocationOnScreen().y
+                + POPUP_OFFSET);
```

Comment:
`getLocationOnScreen()`, `getWidth()` should be called on EDT.

Reply:
Updated. I moved these accessors out of the method and now call them on the EDT and store the value in vars.